components/AutoLogoutProgressBar.tsx

```diff
@@ -0,0 +1,167 @@
+    const [showProgressBar, setShowProgressBar] = useState(false);
+    const [remainingMinutes, setRemainingMinutes] = useState(0);
+    const [completeCalled, setCompleteCalled] = useState(false);
+    const [signal, setSignal] = useState(0);
+
+    const lastInteractionRef = useRef(Date.now());
+
+    useImperativeHandle(ref, () => ({
+        resetTimer: () => {
+            lastInteractionRef.current = Date.now();
+            setSignal(prev => prev + 1);
+            localStorage.setItem("resetLogoutTimer", "X");
+        }
+    }));
+
+    useEffect(() => {
+        const autoLogoutMinutes = props?.autoLogoutMinutes;
+        if (!autoLogoutMinutes) {
+            setShowProgressBar(false);
+            setRemainingMinutes(0);
+            return;
+        }
+
+        setRemainingMinutes(autoLogoutMinutes <= 5 ? autoLogoutMinutes : 5);
+
+        const checkInactivity = () => {
+            const now = Date.now();
+            const inactiveMinutes = (now - lastInteractionRef.current) / 1000 / 60;
+            const minutesLeft = autoLogoutMinutes - inactiveMinutes;
+            setShowProgressBar(minutesLeft <= 5 && minutesLeft > 0);
+        };
+
+        checkInactivity();
+        const interval = setInterval(checkInactivity, 1000);
+        return () => clearInterval(interval);
+    }, [props?.autoLogoutMinutes]);
+
+    const logoutUser = useCallback(() => {
+        localStorage.removeItem("lastClosedAt");
+        logout();
+    }, []);
+
+    const onCompleteFunc = useCallback(() => {
+        setCompleteCalled(true);
+        logoutUser();
+        setTimeout(() => {
+            setCompleteCalled(false);
+        }, 1000);
+    }, []);
+
+    useEffect(() => {
+        if (isReload()) return;
+        if (localStorage.getItem("comesFromEntry") === "true" && props.autoLogoutMinutes) {
+            localStorage.removeItem("lastClosedAt");
+            if (!props.comesFromEntry) {
+                localStorage.setItem("comesFromEntry", "false");
+            }
+            return;
+        }
+        const handleBeforeUnload = () => {
+            const nowIso = new Date().toISOString();
+            if (!localStorage.getItem("lastClosedAt") && !completeCalled) localStorage.setItem("lastClosedAt", nowIso);
+        };
+        window.addEventListener("beforeunload", handleBeforeUnload);
+        const stored = localStorage.getItem("lastClosedAt");
+        if (stored && props.autoLogoutMinutes) {
+            window.removeEventListener("beforeunload", handleBeforeUnload);
+            logoutUser();
+        }
+        return () => {
+            window.removeEventListener("beforeunload", handleBeforeUnload);
+        };
+    }, [completeCalled, props.comesFromEntry]);
+
+    return <>
+        {showProgressBar && <ReverseProgressBar signal={signal} duration={remainingMinutes * 60} className={props.className} label={props.label} onComplete={onCompleteFunc} />}
```

Comment:
i though we do the useImperativeHandle for the ReverseProgressBar reset time to 0 so the duration is set but the remaining propt of the child is set via the handle and the event for reset on window events is handled by the AutoLogoutProgressBar.